controllers/bgppeer_controller_test.go

```diff
@@ -0,0 +1,407 @@
+										"hostA",
+										"hostB",
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			Peer2 := &v1alpha1.BGPPeer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "bgp-peer2",
+					Namespace: MetalLBTestNameSpace,
+				},
+				Spec: v1alpha1.BGPPeerSpec{
+					Address:  "11.0.0.1",
+					ASN:      64001,
+					MyASN:    64000,
+					RouterID: "11.11.11.11",
+				},
+			}
+			By("Creating 1st BGPPeer resource")
+			err := k8sClient.Create(context.Background(), Peer1)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking ConfigMap is created and matches bgp-peer1 configuration")
+			Eventually(func() (string, error) {
+				err := k8sClient.Get(context.Background(),
+					types.NamespacedName{Name: apply.MetalLBConfigMap, Namespace: MetalLBTestNameSpace}, configmap)
+				if err != nil {
+					return "", err
+				}
+				return configmap.Data[apply.MetalLBConfigMap], err
+			}, 2*time.Second, 200*time.Millisecond).Should(MatchYAML(`peers:
+- my-asn: 64500
+  node-selectors:
+  - match-expressions:
+    - key: kubernetes.io/hostname
+      operator: In
+      values:
+      - hostA hostB
+  peer-address: 10.0.0.1
+  peer-asn: 64501
+  router-id: 10.10.10.10
+`))
+			By("Creating 2nd BGPPeer resource")
+			err = k8sClient.Create(context.Background(), Peer2)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking ConfigMap is created and matches bgp-peer1 & 2 configuration")
+			Eventually(func() (string, error) {
+				err := k8sClient.Get(context.Background(),
+					types.NamespacedName{Name: apply.MetalLBConfigMap, Namespace: MetalLBTestNameSpace}, configmap)
+				if err != nil {
+					return "", err
+				}
+				return configmap.Data[apply.MetalLBConfigMap], err
+			}, 2*time.Second, 200*time.Millisecond).Should(MatchYAML(`peers:
+- my-asn: 64500
+  node-selectors:
+  - match-expressions:
+    - key: kubernetes.io/hostname
+      operator: In
+      values:
+      - hostA hostB
+  peer-address: 10.0.0.1
+  peer-asn: 64501
+  router-id: 10.10.10.10
+- my-asn: 64000
+  peer-address: 11.0.0.1
+  peer-asn: 64001
+  router-id: 11.11.11.11
+`))
+
+			By("Deleting 1st BGPPeer resource")
+			err = k8sClient.Delete(context.Background(), Peer1)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking ConfigMap matches bgp-peer2 configuration")
+			Eventually(func() (string, error) {
+				err := k8sClient.Get(context.Background(),
+					types.NamespacedName{Name: apply.MetalLBConfigMap, Namespace: MetalLBTestNameSpace}, configmap)
+				if err != nil {
+					return "", err
+				}
+				return configmap.Data[apply.MetalLBConfigMap], err
+			}, 2*time.Second, 200*time.Millisecond).Should(MatchYAML(`peers:
+- my-asn: 64000
+  peer-address: 11.0.0.1
+  peer-asn: 64001
+  router-id: 11.11.11.11
+`))
+			By("Deleting 2nd BGPPeer resource")
+			err = k8sClient.Delete(context.Background(), Peer2)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking ConfigMap deleted")
+			Eventually(func() (string, error) {
```

Comment:
this can only return err I think no need for the string

Reply:
true since I no longer match configmap string then no need for string removed here and in bgp test